### ISSUE TITLE
First zoom doesn't work when using dateWindow with date objects

### ISSUE THREAD

**Issue by danelkhen** [danvk/dygraphs#568]
When using date objects in dateWindow (instead of date numbers/strings), and marking a section and zooming, I get an empty graph in the first zoom. Only after double clicking and reverting to original size, the second zoom works and shows data correctly. Here's the reproduction code:
```
var data = [
    ["2015-03-25T23:00:00.000Z", 14, 30],
    ["2015-03-25T23:10:00.000Z", 14, 34],
    ["2015-03-25T23:20:00.000Z", 19, 33],
    ["2015-03-25T23:30:00.000Z", 13, 33],
    ["2015-03-25T23:40:00.000Z", 13, 38],
    ["2015-03-25T23:50:00.000Z", 19, 39],
    ["2015-03-26T00:00:00.000Z", 16, 31],
    ["2015-03-26T00:10:00.000Z", 14, 33],
    ["2015-03-26T00:20:00.000Z", 15, 35],
    ["2015-03-26T00:30:00.000Z", 16, 35],
];
var options = {
    dateWindow: [new Date(Date.parse("2015-03-25T23:00:00.000Z")), new Date(Date.parse("2015-03-26T01:00:00.000Z"))],
    //uncommend to make it work again: dateWindow: [Date.parse("2015-03-25T23:00:00.000Z"), Date.parse("2015-03-26T01:00:00.000Z")],
};
data.forEach(function (t) { t[0] = new Date(Date.parse(t[0])); })
var el = document.createElement("div");
document.body.appendChild(el);
var graph = new Dygraph(el, data, options);
```

**Comment by danvk** [comment]
The [docs][1] say:

> dateWindow  
> Initially zoom in on a section of the graph. Is of the form [earliest, latest], where earliest/latest are milliseconds since epoch. **If the data for the x-axis is numeric, the values in dateWindow must also be numbers.**

So the fact that it does something even vaguely reasonable with Date objects could be considered a bug. Just call `.getTime()` on your Date objects to convert them to millis.

[1]: http://dygraphs.com/options.html#dateWindow

**Comment by danelkhen** [comment]
Ahhhh, sorry, I thought that the documentation said that it's possible to use Date objects. Of course I'm using valueOf()/getTime() now, but because it partially worked, I thought it might be an oversight. Thanks for the quick response!

**Comment by tomaszs** [comment]
Well.. after 4 years i can confirm you were right danelkhen that there was something wrong

Here: http://dygraphs.com/tests/dateWindow.html

For date window is used JS Date:

Date.parse("2009/09/29 12:00:00")

I suppose based on this you thought it is possible. I stumbled upon this example just now and also got confused. 

Would be nice to have the example updated to not confuse people. Especially since dygraphs works weird without giving any hint.

Anyways, Dygraphs is still the best :)
